src/quo/components/notifications/count_down_circle.cljs

```diff
@@ -49,60 +48,56 @@
-                                                                       (/ @prev-frame-time 1000))
-                                               current-elapsed      (swap! elapsed-time + delta)
-                                               current-display-time (+ start-at current-elapsed)
-                                               completed?           (>= current-display-time duration)]
-                                           (reset! display-time (if completed?
-                                                                  duration
-                                                                  current-display-time))
-                                           (when-not completed?
-                                             (reset! prev-frame-time frame-time)
-                                             (reset! frame-request (js/requestAnimationFrame
-                                                                    swap-elapsed-time-each-frame))))))]
-    (reagent/create-class
-     {:component-will-unmount #(js/cancelAnimationFrame @frame-request)
-      :reagent-render
-      (fn []
-        (reset! frame-request (js/requestAnimationFrame swap-elapsed-time-each-frame))
-        [rn/view
-         {:style {:position :relative
-                  :width    size
-                  :height   size}}
-         [svg/svg
-          {:view-box (str "0 0 " size " " size)
-           :width    size
-           :height   size}
-          [svg/path
-           {:d path :fill :none :stroke (or trail-color :transparent) :stroke-width stroke-width}]
-          (when-not (= @display-time duration)
-            [svg/path
-             {:d                 path
-              :fill              :none
-              :stroke            (or color (get-in themes [:color theme]))
-              :stroke-linecap    :square
-              :stroke-width      stroke-width
-              :stroke-dasharray  path-length
-              :stroke-dashoffset (linear-ease @display-time 0 path-length duration)}])]])})))
-
-(def circle-timer (quo.theme/with-theme circle-timer-internal))
+(defn circle-timer
+  [{:keys [color duration size stroke-width trail-color rotation initial-remaining-time]}]
+  (let [theme                           (quo.theme/use-theme-value)
+        rotation                        (or rotation :clockwise)
+        duration                        (or duration 4)
+        stroke-width                    (or stroke-width 1)
+        size                            (or size 9)
+        max-stroke-width                (max stroke-width 0)
+        {:keys [path path-length]}      (get-path-props size max-stroke-width rotation)
+        start-at                        (get-start-at duration initial-remaining-time)
+        elapsed-time                    (rn/use-ref-atom 0)
+        prev-frame-time                 (rn/use-ref-atom nil)
+        frame-request                   (rn/use-ref-atom nil)
```

Comment:
why is frame-request now a ref? it should be state, no?
same for many of these uses?

or was initially implemented wrong?

Reply:
rechecked again, and it should work fine with atom no need in re-rendering